java/feature-layer-shapefile/src/main/java/com/esri/arcgisruntime/sample/featurelayershapefile/MainActivity.java

```diff
@@ -34,6 +34,8 @@
   private final static String TAG = MainActivity.class.getSimpleName();
 
   private MapView mMapView;
+  // objects that implement Loadable must be class fields to prevent being garbage collected before loading
+  private ShapefileFeatureTable mShapefileFeatureTable;
```

Comment:
This doesn't have to be a member variable because it's attached to the feature layer outside of a done loading listener

Reply:
I'm unsure here: it looks to me that there's a reference to the feature table inside the done loading listener, in that if the scope were to be lost the user might not get an error back from it on line 66?

Reply:
Oh yeah sorry, these lines should be outside of the doneLoadingListener:

```
 // add the feature layer to the map
        mMapView.getMap().getOperationalLayers().add(shapefileFeatureLayer);
```

Then the table doesn't have to be global, I believe

Reply:
my bad that was sloppy of me